### DISCUSSION TITLE
Classification on a static graph - only node and edge features change

### DISCUSSION THREAD

**Discussion by magottsc** [pyg-team/pytorch_geometric#4726] [Q&A]
Hey,

I have a static graph (node configuration doesn't change) and have to predict the category of all nodes (5 categories) given their changing node / edge features
So the only thing that changes in my dataset are node and edge features
In the end it looks like a pretty standard supervised learning classification problem
I have looked at multiple angles now from node to graph classification and nothing really fits this problem
Node classification is more aimed at classifying an "unknown" node in a graph, graph classification is more about classifying graphs of different size / configuration
The problem I have to solve is fairly "nodal" so solving it with a graph approach is the preferred way
Please does anyone have a suggestion on where to start, I seem a bit lost currently

Thank you

**Comment by Padarn** [comment]
I think node classification fits this problem fairly well? I don't think its that important that your graph doesn't change.

**Comment by rusty1s** [comment]
If you have a static graph with a sequence of changing node and edge features, I think the general approach to tackle this problem is to apply a GNN for each timestamp, and then merge node features across timestamps with an LSTM/Transformer. Alternatively, one could think of constructing a graph that is augmented by temporal edges, i.e., you have a snapshot of the graph at each timestamp, and you add additional edges to nodes to previous and later timestamps.

**Comment by magottsc** [comment]
Thanks, that's helpful

In my example I have certain forecasts as features for every node, e.g. temperature and windspeed etc. which are related temporally
So i believe it makes sense to construct a heterogenous graph with temporal edges and spacial edges with features within one timestep

In my case i'd have a graph of 24 timesteps, representing 24 hours of a day
I need to make a prediction based on the node / edge features (forecasts for every hour of the day) for the next 24 hours

And in the end i need to be able to categorize for every hour (5 categories per node)

My plan for now:
a) I would go ahead and initally design a spatio temporal graph with 24 timesteps
b) Then try and design a Neural Net around my needs (has anyone seen anything similar that I could use for inspiration)?

**Reply by Padarn** [reply]
I'd suggest you take a look at https://github.com/benedekrozemberczki/pytorch_geometric_temporal - there are some similar models in pytorch geometric too, but this library has more examples so you might find something matching what you need exactly.

However I concur with @rusty1s - I think usually instead of constructing a heterogeneous graph, people would normally use something like a recurrent neural network to handle the time dimension. Maybe something like https://pytorch-geometric-temporal.readthedocs.io/en/latest/_modules/torch_geometric_temporal/nn/recurrent/temporalgcn.html#TGCN
